server/src/endpoints/orderNewDrink.ts

```diff
@@ -0,0 +1,57 @@
+import { AuthenticatedEndpointFunction, LogFn, Message } from '../endpoint'
+import { minimizeUser, updateUserProfile, User } from '../user'
+import generators from '../generators'
+
+const orderNewDrink: AuthenticatedEndpointFunction = async (user: User, inputs: any, log: LogFn) => {
+  let thisDrinkSkeleton: string
+
+  let privateActionString: string
+
+  const oldItem = user.item
+
+  const drinkVessels = generators.drinkVessels
+  const drinkContents = generators.drinkContents
+  const drinkNames = generators.drinkNames
+  const drinkSkeletons = generators.drinkSkeletons
+
+  const thisDrinkVessel = drinkVessels.generate()
+  const thisDrinkContent = drinkContents.generate()
+  const thisDrinkName = drinkNames.generate()
+  /* These three variables need to be put into the database to be retrieved when someone orders an existing drink. */
+  thisDrinkSkeleton = drinkSkeletons.generate()
+
+  thisDrinkSkeleton = thisDrinkSkeleton.replace('-drinkVessel-', thisDrinkVessel)
+  thisDrinkSkeleton = thisDrinkSkeleton.replace('-drinkContent-', thisDrinkContent)
+  thisDrinkSkeleton = thisDrinkSkeleton.replace('-drinkName-', thisDrinkName)
+  thisDrinkSkeleton = thisDrinkSkeleton.replace('-userName-', user.realName)
+
+  const item = thisDrinkName
+
+  const actionString = (item ? `picks up ${item}.` : `drops ${oldItem}.`)
+
+  const newProfile = await updateUserProfile(user.id, { item })
+
+  const messages: Message[] = [
+    {
+      userId: user.roomId,
```

Comment:
For `actionString` and `thisDrinkSkeleton`, which ones are supposed to be private to just the player getting a drink and which ones are supposed to show to everyone in the room?

For "so-and-so is drinking a potion" commands, you probably want to use `emote` instead of `privateCommand` (and target the roomId). For a private "you are doing a thing" message, `privateCommand` does work, but you want the `userId` to be e.g. `user.id` instead of `user.roomId`. Sending it to `user.roomId` will send it to every user in the room instead of just that one person.

Reply:
They're both supposed to go to everyone in the room. However, probably 'you are doing this thing' can just be private, I'll change it accordingly. Should I change thisDrinkSkeleton? I do want everyone to see that one.

Reply:
Would chatMessage be a more appropriate target?

Reply:
"chatMessage" is what we use for a normal text chat message you've sent, that probably isn't appropriate.

What do you want this to look like?

Reply:
I want it to look like a thing happening in the room? I don't know what would be right for that.

Reply:
"A thing happening in the room" = an italicized message that isn't a specific player doing an action?

I think in that case `privateCommand` sent to the entire room is the only tool we have, even though it's named incorrectly / there's no guarantee it will continue to look that way in the future since this isn't what it's intended to be used for.

Reply:
I've been looking through the other endpoints and haven't found one that seems to do something similar? The closest I've found is something like displayMessage for when talking to the bartender, but that's a private command.

Reply:
Yea, that's what I'm looking for it to be. Changed to to privateCommand for now.